src/main/java/org/broadinstitute/hellbender/utils/iterators/IntervalAlignmentContextIterator.java

```diff
@@ -0,0 +1,109 @@
+package org.broadinstitute.hellbender.utils.iterators;
+
+import htsjdk.samtools.SAMSequenceDictionary;
+import org.broadinstitute.hellbender.engine.AlignmentContext;
+import org.broadinstitute.hellbender.utils.IntervalUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.pileup.ReadPileup;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * For special cases where we want to emit AlignmentContexts regardless of whether we have an overlap with a given interval.
+ *
+ * In other words, this will iterate over each base of each interval and emit an alignment context.  If no reads overlap,
+ *  it will emit an empty alignment context.  Each empty alignment context will be a new instance.
+ */
+public class IntervalAlignmentContextIterator implements Iterable<AlignmentContext>, Iterator<AlignmentContext> {
```

Comment:
I agree with @vruano, Iterable means that it is a collection.  The interface has spliterator and forEach.

If LocusWalker (and LocusWalkerSpark) can handle doing an iterator and forEachRemaining call, instead (of spliterator and forEach).  Under the hood, the spliterator call here is just grabbing an iterator anyway.

The problem that I see is that we may require a spliterator in the future.  But to solve this, I would rather just instantiate the spliterator explicitly from the iterator (``Spliterators.spliteratorUnknownSize(iterator, Spliterator.ORDERED)``).  That way, @vruano is satisfied and we keep the flexibility of spliterator.  In another branch I have a Factory class, which I can leverage to encapsulate away this logic.